### PR TITLE
feat(ingest/kakfa-connect): support generic connectors

### PR DESCRIPTION
**Pull request by BoyuanZhangDE**
Make kafka-connect ingestion support connectors other than confluent JDBC connector and debezium connector

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Comment by BoyuanZhangDE** [comment]
Make kafka-connect ingestion support connectors other than confluent JDBC connector and debezium connector

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py [hunk 1]
```diff
@@ -67,6 +73,10 @@ class KafkaConnectSourceConfig(DatasetLineageProviderConfigBase):
         default=None,
         description='Platform instance mapping to use when constructing URNs. e.g.`platform_instance_map: { "hive": "warehouse" }`',
     )
+    generic_connectors: Optional[List[GenericConnectorConfig]] = Field(
+        default=None,
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:77]:
```suggestion
    generic_connectors: List[GenericConnectorConfig] = Field(
        default=[]],
```

File: metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py [hunk 2]
```diff
@@ -923,20 +934,47 @@ def get_connectors_manifest(self) -> List[ConnectorManifest]:
...
                 else:
-                    # Debezium Source Connector lineages
-                    try:
-                        connector_manifest = DebeziumSourceConnector(
-                            connector_manifest=connector_manifest, config=self.config
-                        ).connector_manifest
+                    # empty list of customized connector
+                    if not self.config.generic_connectors:
+                        logger.warning(
+                            f"Detected undefined connector {connector_manifest.name}, but user defined no customized connectors. Please refer to Kafka Connect ingestion recipe to define this customized connector."
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/kafka_connect.py:947]: this error checking becomes redundant given the check below, once the default is changed to []

**Review by hsheth2 — CHANGES_REQUESTED** [review]
Looking good.

If possible, it'd be great to add some tests for this as well :)

**Comment by BoyuanZhangDE** [comment]
> Looking good.
> 
> If possible, it'd be great to add some tests for this as well :)

will do!

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**